OMCompiler/Compiler/runtime/OMSimulator_omc.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of OpenModelica.
+ *
+ * Copyright (c) 1998-2010, Linköpings University,
+ * Department of Computer and Information Science,
+ * SE-58183 Linköping, Sweden.
+ *
+ * All rights reserved.
+ *
+ * THIS PROGRAM IS PROVIDED UNDER THE TERMS OF THIS OSMC PUBLIC
+ * LICENSE (OSMC-PL). ANY USE, REPRODUCTION OR DISTRIBUTION OF
+ * THIS PROGRAM CONSTITUTES RECIPIENT'S ACCEPTANCE OF THE OSMC
+ * PUBLIC LICENSE.
+ *
+ * The OpenModelica software and the Open Source Modelica
+ * Consortium (OSMC) Public License (OSMC-PL) are obtained
+ * from Linköpings University, either from the above address,
+ * from the URL: http://www.ida.liu.se/projects/OpenModelica
+ * and in the OpenModelica distribution.
+ *
+ * This program is distributed  WITHOUT ANY WARRANTY; without
+ * even the implied warranty of  MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
+ * IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS
+ * OF OSMC-PL.
+ *
+ * See the full OSMC Public License conditions for more details.
+ *
+ */
+
+
+#include <stdio.h>
+
+#if defined(__MINGW32__)
+#  include <windows.h>
+#else
+#  include <dlfcn.h>
+#endif
+
+#include "meta_modelica.h"
+#include <iostream>
+using namespace std;
+
+extern "C" {
+
+//#include "OMSimulator.h"
+#include "ModelicaUtilities.h"
+
+# if defined WIN32
+   static HINSTANCE OMSimulatorDLL;
+   #define AddressOf GetProcAddress
+# else
+   static void * OMSimulatorDLL;
+   #define AddressOf dlsym
+# endif
+/*
+# if defined WIN32
+  static HINSTANCE OMSimulatorDLL;
+# else
+  static void * OMSimulatorDLL;
+# endif
+*/
+void loadDLL()
+{
+  // check for OMSimulatorDLL instance for the first time this function called and load the library
+	if (!OMSimulatorDLL) {
+		#if defined WIN32
+			OMSimulatorDLL = LoadLibrary("libOMSimulator.dll");
+		#else
+			OMSimulatorDLL = dlopen("libOMSimulator.so",RTLD_LAZY);
+		#endif
+		if (!OMSimulatorDLL) {
+    	cout << "Could not load the dynamic library libOMSimulator, Exiting the program";
+     	exit(0);
+  	}
+  }
+}
+
+extern const char* OMSimulator_getVersion()
+{
+  loadDLL();
+  //const char * res=oms_getVersion();
+	//printf("%s",res);
+	typedef char* (*fnptr_oms_getVersion)();
+  fnptr_oms_getVersion oms_getVersion = (fnptr_oms_getVersion)AddressOf(OMSimulatorDLL, "oms_getVersion");
```

Comment:
I would suggest to resolve the symbols in the loadDLL function, so it is only done once.